cocotb/result.py

```diff
@@ -36,6 +36,10 @@
 def raise_error(obj, msg):
     """Creates a :exc:`TestError` exception and raises it after printing a traceback.
 
+    Note that since cocotb will print a traceback for you when an exception is
+    caught, you are unlikely to need this - use ``raise TestError(msg)``
+    instead.
+
```

Comment:
Can we add a `.. deprecated:: 1.3` here (http://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-deprecated), and possibly throw a DeprecationWarning`?

Same applies for the `create_error` function below.

Reply:
Sure, I suppose there's no reason not to do that. Updated with tests.